### PR TITLE
Refactoring/buffer to extension methods

### PR DESCRIPTION
**Pull request by DaEgi01**
see both individual commits.
take care in places where the id is updated in a loop.
I almost fucked it up in one place before I realized what is going on :D

### CODE REVIEW

**Comment by kianzarrin** [comment]
image
when I create a new branch I do a fetch first and then base it on the remote origin/master. (i guess pull also works)
image

File: TLM/TLM/UI/TrafficManagerTool.cs [hunk 1]
```diff
@@ -1848,23 +1848,22 @@ void DefaultRenderOverlay(RenderManager.CameraInfo cameraInfo)
         // TODO: Extract into a Debug Tool GUI class
         private void DebugGuiDisplayBuildings() {
             GUIStyle _counterStyle = new GUIStyle();
-            BuildingManager buildingManager = Singleton<BuildingManager>.instance;
 
             for (int i = 1; i < BuildingManager.MAX_BUILDING_COUNT; ++i) {
```
Comment (kianzarrin) [inline TLM/TLM/UI/TrafficManagerTool.cs:1844]: i could be renamed to buildingId

**Review by kianzarrin — APPROVED** [review]

**Review by krzychu124 — APPROVED** [review]
cool 👍

File: TLM/TLM/UI/TrafficManagerTool.cs (hunk 1 above)
Comment (DaEgi01) [reply TLM/TLM/UI/TrafficManagerTool.cs:1844]: will do

**Pull request merged**